docs/BandOracle.md

```diff
@@ -0,0 +1,313 @@
+
+---
+
+### `removeSymbol()`
+
+```cadence
+fun removeSymbol(symbol: String)
+```
+Auxiliary private function for removing a stored symbol
+
+Parameters:
+  - symbol : _The string representing the symbol to delete_
+
+---
+
+### `_getRefData()`
+
+```cadence
+fun _getRefData(symbol: String): RefData?
+```
+Auxiliary private function for checking and retrieving data for a given symbol.
+
+Parameters:
+  - symbol : _String representing a symbol._
+
+Returns: Optional `RefData` struct if there is any quote stored for the requested symbol.
+
+---
+
+### `calculateReferenceData()`
+
+```cadence
+fun calculateReferenceData(baseRefData: RefData, quoteRefData: RefData): ReferenceData
+```
+Private function that calculates the reference data between two base and quote symbols.
+
+Parameters:
+  - baseRefData : _Base ref data._
+  - quoteRefData : _Quote ref data._
+
+Returns: Calculated `ReferenceData` structure.
+
+---
+
+### `setFee()`
+
+```cadence
+fun setFee(fee: UFix64)
+```
+Private method for the `FeeCollector` to be able to set the fee for using the oracle
+
+Parameters:
+  - fee : _The amount of flow tokens to set as fee._
+
+---
+
+### `collectFees()`
+
+```cadence
+fun collectFees(): FungibleToken.Vault
+```
+Private method for the `FeeCollector` to be able to collect the fees from the contract vault.
+
+Returns: A flow token vault with the collected fees so far.
+
+---
+
+### `getReferenceData()`
+
+```cadence
+fun getReferenceData(baseSymbol: String, quoteSymbol: String, payment: FungibleToken.Vault): ReferenceData
+```
+Public access functions.
+The entry point for consumers to query the oracle in exchange of a fee.
+
+Parameters:
+  - baseSymbol : _String representing base symbol._
+  - quoteSymbol : _String representing quote symbol._
+  - payment : _Flow token vault containing the service fee._
+
+Returns: The `ReferenceData` containing the requested data.
+
+---
+
+### `createRelay()`
+
+```cadence
+fun createRelay(updaterCapability: Capability<&{DataUpdater}>): Relay
+```
+Public method for creating a relay and become a relayer.
+
+Parameters:
+  - updaterCapability : _The capability pointing to the OracleAdmin resource needed to create the relay._
+
+Returns: The new relay resource.
+
+---
+
+### `getUpdaterCapabilityNameFromAddress()`
+
+```cadence
+fun getUpdaterCapabilityNameFromAddress(relayer: Address): String
+```
+Auxiliary method to ensure that the formation of the capability name that
+identifies data updater capability for relayers is done in a uniform way
+by both admin and relayers.
+
+Parameters:
+  - relayer : _Address of the account who will be granted with a relayer._
+
+---
+
+### `getFee()`
+
+```cadence
+fun getFee(): UFix64
+```
+This function returns the current fee for using the oracle in Flow tokens.
+
+Returns: The fee to be charged for every request made to the oracle.
+
+---
+
+### `getFreeReferenceData()`
+
+```cadence
+fun getFreeReferenceData(baseSymbol: String, quoteSymbol: String): ReferenceData?
```

Comment:
Same question about whether the free access is intended for release

Reply:
this one wasn't on purpose, I'm going to re-generate the docs once the example consumer contract is done on this branch https://github.com/onflow/band-oracle-contracts/tree/alilloig/docs-examples